blockchain/My-Wallet-V3-iOS
# feat(KYC): IOS-1167 attach authorization token to KYC personal details screen

## Objective

Complete personal detail screen

## Description

Completing the personal details screen—previously, the network call made after tapping on "next" did not work because it was blocked by KYC authentication.

Other changes:
* Creating `KYCPageViewFactory` which is in charge of constructing a `KYCBaseViewController` given a `KYCPageType` and `KYCPayload`
* Creating `KYCPayload` and updating the `nextPageFromPageType`. The idea behind this is that when a page completes it can pass in a `KYCPayload` (e.g. thought of this so that we can pass around the data selected from `KYCCountrySelectionController`

## How to Test

1. Pull these changes in
2. Add `|| host == "api.dev.blockchain.info"` to `NetworkManager`
3. Launch KYC -> Select "Germany" -> Fill out personal details screen -> verify that you can get passed this step

NOTE: if you fail to get a `KYCUser` when you create a new wallet, _make sure you use a unique email address._ This is because the backend will complain if you try to create a new `KYCUser` if one has already been created for the email address provided (an email address already exists for the address "test@doesnotexist.com")

## Screenshot

## Related Information

* Ticket Number: IOS-1167

## Merge Checklist
